providers/yttcb/pinniped_addon_secret.lib.yaml

```diff
@@ -0,0 +1,204 @@
+      tlsKey: /etc/dex/tls/tls.key
+    expiry:
+      signingKeys: 90m
+      idTokens: 5m
+      authRequests: 90m
+      deviceRequests: 5m
+    logger:
+      level: info
+      format: json
+    staticClients:
+      - id: "pinniped" #! Do not change. Will be updated by post-deployment job
+        redirectURIs:
+          - https://0.0.0.0/callback #! Do not change. Will be updated by post-deployment job
+        name: "pinniped" #! Do not change. Will be updated by post-deployment job
+        secret: "dummyvalue" #! Do not change. Will be updated by post-deployment job
+    ldap:
+      host: #@ data.values.LDAP_HOST
+      insecureNoSSL: false
+      startTLS: null
+      rootCA: null #! <rootCA> or <rootCAData> if required when LDAP host is using self signed certificate. Path to the CA file
+      rootCAData: #@ data.values.LDAP_ROOT_CA_DATA_B64
+      bindDN: #@ data.values.LDAP_BIND_DN
+      BIND_PW_ENV_VAR: #@ data.values.LDAP_BIND_PASSWORD
+      usernamePrompt: LDAP Username
+      insecureSkipVerify: false
+      userSearch:
+        baseDN: #@ data.values.LDAP_USER_SEARCH_BASE_DN
+        filter: #@ data.values.LDAP_USER_SEARCH_FILTER
+        username: #@ data.values.LDAP_USER_SEARCH_USERNAME
+        idAttr: #@ data.values.LDAP_USER_SEARCH_ID_ATTRIBUTE
+        emailAttr: #@ data.values.LDAP_USER_SEARCH_EMAIL_ATTRIBUTE
+        nameAttr: #@ data.values.LDAP_USER_SEARCH_NAME_ATTRIBUTE
+        scope: sub
+      groupSearch:
+        baseDN: #@ data.values.LDAP_GROUP_SEARCH_BASE_DN
+        filter: #@ data.values.LDAP_GROUP_SEARCH_FILTER
+        nameAttr: #@ data.values.LDAP_GROUP_SEARCH_NAME_ATTRIBUTE
+        scope: sub
+        userMatchers: #! Add more matchers if necessary
+          - userAttr: #@ data.values.LDAP_GROUP_SEARCH_USER_ATTRIBUTE
+            groupAttr: #@ data.values.LDAP_GROUP_SEARCH_GROUP_ATTRIBUTE
+    oauth2:
+      skipApprovalScreen: true
+      responseTypes:
+        - "code"
+    storage:
+      type: kubernetes
+      config:
+        inCluster: true
+    enablePasswordDB: false
+  service: #@ getServiceValuesForMC(dex_service_name)
+  deployment:
+    replicas: 1
+  certificate: #! required
+    duration: #@ data.values.CERT_DURATION
+    renewBefore: #@ data.values.CERT_RENEW_BEFORE
+  dns: #! required
+    vsphere:
+      dnsNames:
+        - tkg-dex.com
+      ipAddresses: [ 0.0.0.0 ] #! Do not change. Will be updated by post-deployment job
+    aws:
+      dnsNames:
+        - tkg-dex.com
+      DEX_SVC_LB_HOSTNAME: example.com #! <DEX_SVC_LB_HOSTNAME> is required for AWS
+    azure:
+      dnsNames:
+        - tkg-dex.com
+      DEX_SVC_LB_HOSTNAME: dex.example.com #! <DEX_SVC_LB_HOSTNAME> is required for azure
+#@ end
+
+#@ def getValuesForWC():
+#@ bomData = get_bom_data_for_tkr_name()
+#@ pinnipedImage = bomData.components["pinniped"][0].images.pinnipedImage
+#@ postDeployPinnipedImage = bomData.components["tanzu-framework-addons"][0].images.tkgPinnipedPostDeployImage
```

Comment:
It appears, from our testing, that the shape of the `tkr-bom` has changed since this bit of code was originally written. This works in `tkg 1.6.0` but in the `1.7.0 classy` cluster I am experimenting with, the following change is required:

```yaml
# old bad
#! postDeployPinnipedImage = bomData.components["tanzu-framework-addons"][0].images.tkgPinnipedPostDeployImage
# new good
#@ postDeployPinnipedImage = bomData.components["pinniped-post-deploy"][0].images.tkgPinnipedPostDeployImage
```

Reply:
Pushing an update for this.